### ISSUE TITLE
toHaveStyleRule optional 'supports' Option has no effect

### ISSUE THREAD

**Issue by leotm** [styled-components/jest-styled-components#230]
```jsx
const Button = styled.button`
  @media (max-width: 640px) {
    &:hover {
      color: red;
    }
  }
`
```

```jsx
test('it works', () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toHaveStyleRule('color', 'red', {
    media: '(max-width:640px)',
    modifier: ':hover',
    supports: '(display: grid) ',
  })
})
```

Should fail with:

```
No style rules found on passed Component using options
{"media":"(max-width: 640psx)","modifier":":hover","supports":"(display: grid)"}
```

But passes as a false positive.
